charmstore.go

```diff
@@ -359,11 +359,21 @@ func (s *CharmStore) GetResource(id *charm.URL, revision int, name string) (resu
 	}, nil
 }
 
-// Publish tells the charmstore to mark the given charm as published with the given resource revisions.
-func (s *CharmStore) Publish(id *charm.URL, resources map[string]int) (*charm.URL, error) {
+// Publish tells the charmstore to mark the given charm as published with the
+// given resource revisions to the given channels. If no channels are specified,
+// the stable channel will be assumed.
+func (s *CharmStore) Publish(id *charm.URL, channels []string, resources map[string]int) (*charm.URL, error) {
```

Comment:
The point of the wrapper is to hide the ugly details of the HTTP API.  The params package should be an implementation detail, IMO... that way you're not letting the API details leak outside the wrapper.  Given that you're generally going to be using this from a CLI where we have to translate between a plain string and the API type anyway... seems like we might as well hide it away as far as possible.